lib/engine/game/base.rb

```diff
@@ -592,8 +599,32 @@ def initialize_actions(actions)
         @loading = false
       end
 
+      def process_derived_action(action)
+        action = action_from_h(action) if action.is_a?(Hash)
+        action.derived = true
+
+        @action_queue.prepend(action)
+      end
+
       def process_action(action)
         action = action_from_h(action) if action.is_a?(Hash)
+        raise GameError, "Derived actions may not use process_action, especially #{action}" if action.derived
+
+        # Process the action we came here to do first
+        result = process_single_action(action)
+
+        # Process any other actions put on the queue since our action
+        until @action_queue.empty?
+          derived_action = @action_queue.pop
```

Comment:
If it's a queue, it should be `shift`. `pop` is for stacks (FIFO vs FILO)

Reply:
If this PR goes anywhere I need to fix this, and test it, before it can get merged